python/ray/dashboard/dashboard.py

```diff
@@ -701,6 +718,109 @@ def run(self):
             if counter % 10:
                 self.update_nodes()
 
+class TuneCollector(threading.Thread):
+
+
+    def __init__(self, logdir, logger, reload_interval):
+        """Initialize collector worker thread.
+        Args
+            logdir (str): Directory path to save the status information of
+                          jobs and trials.
+            logger (Logger): Logger for collector thread.
+        """
+
+        super().__init__()
+        self._logdir = logdir
+        self._trial_records = {}
+        self.logger = logger
+        self._data_lock = threading.Lock()
+        self._reload_interval = reload_interval
+        self._available = False
+
+    def get_stats(self):
+        with self._data_lock:
+            return {
+                "trial_records": copy.deepcopy(self._trial_records)
+            }
+
+    def get_availability(self):
+        with self._data_lock:
+            return {
+                "available": self._available
+            }
+
+    def run(self):
+        while True:
+            with self._data_lock:
+                self.collect()
+            time.sleep(self._reload_interval)
+
+    def collect(self):
```

Comment:
Can you leave a docstring as to what you're doing here? 

Also consider moving some of the cleaning code to a helper function to improve readability.

Reply:
+1